### DISCUSSION TITLE
coreboot on the lite mkIII

### DISCUSSION THREAD

**Discussion by ghost** [StarLabsLtd/firmware#130] [General] [answered]
So yesterday I was exploring how to swtich-branch on the lite mk3 from AMI to coreboot, and I could not resist and ended up switching despite it being in beta yet. I have version 8.37. This version has a few bugs like only powering on when with the power cord. How if possible someone can contribute to help make it better?

**Comment by Sean-StarLabs** [comment]
Everything is kept in coreboots gerrit if you want to contribute

**Comment by ghost** [comment]
I was checking in the LVFS website and it says the liscense of the coreboot of starlabs is proprietary. Is it correct? https://fwupd.org/lvfs/devices/com.starlabs.L3.coreboot
![immagine](https://github.com/StarLabsLtd/firmware/assets/146669079/5cdc8bb1-791e-4212-acfc-a8dbda17b4ac)

**Reply by Sean-StarLabs** [reply]
Yup

**Reply by ghost** [reply]
Are there plans to make it open-source with some GPL or MIT liscense? Can you explain why it has a proprietary liscense?

**Reply by Sean-StarLabs** [reply]
I think you're thinking of the term "proprietary firmware", not the license. Most of it is already open source.

**Reply by ghost** [reply]
So why doesn't it have a open-source liscence like GPL? Is it because of the part is not open source yet? I ask for curiosity, and of course would want it to be 100% open source, but would be glad to understand why that's not the case yet.

**Reply by Sean-StarLabs** [reply]
Most of it does - but as a whole, it'll always be proprietary.

**Reply by ghost** [reply]
for legal reasons?

**Reply by Sean-StarLabs** [reply]
Not really, it's just following the way coreboot does it (save for the blobs)

**Reply by ghost** [reply]
I see, so the ultimate foss alternative would be libreboot afaik it. So that's about the firmware coreboot, thanks for clarifying. What about the reasons of why the starlabs license of the coreboot is proprietary?

**Reply by Sean-StarLabs** [reply]
We don't have a license of coreboot, it's inherited

**Reply by ghost** [reply]
I don't understand, coreboot has a GPL license according to their statement: https://github.com/coreboot/coreboot#licenses

**Reply by Sean-StarLabs** [reply]
Correct - coreboot is proprietary and licensed under GPL 2

**Reply by ghost** [reply]
I don't understand why you say that coreboot is proprietary while using a GPL2 which requires the source code to be open and available, including those who use its source code to fork it or whatever to remain open and available.
https://github.com/coreboot/coreboot/blob/main/COPYING

**Reply by Sean-StarLabs** [reply]
What do you think proprietary means?

**Reply by ghost** [reply]
That it does not have its source code open, to by shared, modified copied forked ecc.

**Reply by Sean-StarLabs** [reply]
It means that the code is owned. There's a common term, proprietary firmware, used to refer to stock firmware on computers from the likes of AMI, Phoenix etc. They're usually closed (less so than people think) but still, the terms get blurred.

**Reply by ghost** [reply]
But coreboot is different from those closed stock firmwares, then not proprietary. It's open. The coreboot version developed by starlabs does it keep the same GLP2 license of coreboot?

**Reply by Sean-StarLabs** [reply]
Proprietary does not mean closed.

We don't have our own version of coreboot; but yes, coreboot is GPL 2 licensed.

**Reply by ghost** [reply]
I mean the implementation of coreboot by starlabs, does starlabs add any piece of code that is not open?

**Reply by Sean-StarLabs** [reply]
To the firmware binaries, yes

**Reply by ghost** [reply]
Are those the blobs necessary for wifi, Bluetooth and etc, or something else?

**Reply by Sean-StarLabs** [reply, marked as answer]
Yes, for basic operation - they are for almost all computers made in the last ~15 years or so

**Reply by ghost** [reply]
Thanks for clarifying all my questions. It's great to have this discussion session here.
